### ISSUE TITLE
Consider lifetime outlives relation when resolving multiple borrows

### ISSUE THREAD

**Issue by voidc** [xldenis/creusot#824]
When multiple locals must be resolved at the same program point, we currently use their index as a heuristic for the correct order of resolution. A more correct solution would be to consider the outlives relation between the borrows' lifetimes. This information could be obtained using the [`RegionInferenceContext::outlives_constraints`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_borrowck/region_infer/struct.RegionInferenceContext.html#method.outlives_constraints) method.

**Comment by xldenis** [comment]
@jhjourdan is this still an issue?

**Comment by jhjourdan** [comment]
Yes, this is still an issue, I haven't changed anything wrt. the order in which things get resolved.

**Comment by jhjourdan** [comment]
@lafeychine and I stumbled upon that while working on the MPMC queue.
